### PR TITLE
feat: add editDate field support to Qt client

### PR DESCRIPTION
**Pull request by ckerr**
Adds support for the editDate field.

Refactors the Torrent class properties, adding a bitfield to know which properties change as a result of `Torrent::update()`

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -22,3 +22,7 @@
 	path = third-party/miniupnpc
 	url = https://github.com/transmission/miniupnpc
 	branch = post-2.0.20170509-transmission
+[submodule "third-party/units"]
```
Comment (mikedld) [inline .gitmodules]: Not convinced the amount and type of operations we do on unit-based variables warrants this dependency.

File: qt/DetailsDialog.cc [hunk 2]
```diff
@@ -164,7 +165,7 @@ class PeerItem : public QTreeWidgetItem
                 if (ipAddress.protocol() == QAbstractSocket::IPv4Protocol)
                 {
                     quint32 const ipv4Address = ipAddress.toIPv4Address();
-                    collatedAddress = QLatin1String("1-") + QString::fromLatin1(QByteArray::number(ipv4Address, 16).
+                    collatedAddress = QLatin1String("1-") % QString::fromLatin1(QByteArray::number(ipv4Address, 16).
```
Comment (mikedld) [inline qt/DetailsDialog.cc]: This (and lots of other stuff) is not something related directly to editDate support, I'd rather split this PR in two if you want to make additional refactoring preceding or following the advertised changes.

File: qt/FilterBar.cc [hunk 3]
```diff
@@ -232,9 +232,16 @@ FilterBar::FilterBar(Prefs& prefs, TorrentModel const& torrents, TorrentFilter c
     connect(&myTorrents, SIGNAL(modelReset()), this, SLOT(recountSoon()));
     connect(&myTorrents, SIGNAL(rowsInserted(QModelIndex, int, int)), this, SLOT(recountSoon()));
     connect(&myTorrents, SIGNAL(rowsRemoved(QModelIndex, int, int)), this, SLOT(recountSoon()));
-    connect(&myTorrents, SIGNAL(dataChanged(QModelIndex, QModelIndex)), this, SLOT(recountSoon()));
     connect(myRecountTimer, SIGNAL(timeout()), this, SLOT(recount()));
 
+    connect(&myTorrents, &TorrentModel::torrentsChanged, [this](torrent_ids_t const& /*ids*/, Torrent::fields_t const& fields)
```
Comment (mikedld) [inline qt/FilterBar.cc]: In light of recent fixes in #1107, please pass `this` as context to trigger signal disconnects on object destruction.

**Comment by ckerr** [comment]
I'm going to break this PR into a couple of different parts to keep each PR more focused. First one is at https://github.com/transmission/transmission/pull/1234

**Comment by RobFog** [comment]
Should this be removed from the 3.0x milestone?

**Pull request closed without merging**